### ISSUE TITLE
False positive on SC2185 (find without path argument)

### ISSUE THREAD

**Issue by vog** [koalaman/shellcheck#1748]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2185
- My shellcheck version (`shellcheck --version` or 'online'): online and 0.5.0
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
find --version
```

#### Here's what shellcheck currently says:

```
SC2185: Some finds don't have a default path. Specify '.' explicitly.
```
#### Here's what I wanted or expected to see:

No warning at all.

Rationale: Calling "find --version", or "SOMETOOL --version" is the standard way of checking whether the tool is properly installed or not. Portable shell scripts usually have a bunch of those commands at the very beginning, to avoid spurious failure on minimalistic (and/or embeeded) system. As such, an execution with "--version", "--help" and similar options should not trigger the warning of a missing path argument.

**Comment by Huanzo** [comment]
I would agree that this should not throw SC2185.
But it should throw another lint for using `--version` or `-v` for checking the existence of a command.

The POSIX way of checking for the existence of a command would be to use `command -v COMMANDNAME`

**Comment by gromgit** [comment]
While I agree with @Huanzo that `<cmd> --version` is not "the standard way of checking whether the tool is properly installed or not", I think that SC2185 _should_ be thrown, as not every userland is a GNU userland. BSD/macOS `find`, for instance, really doesn't like this:
```
macOS$ /usr/bin/find -type f
/usr/bin/find: illegal option -- t
usage: find [-H | -L | -P] [-EXdsx] [-f path] path ... [expression]
       find [-H | -L | -P] [-EXdsx] -f path [path ...] [expression]
```
You can see from the help text that paths are mandatory.

**Comment by vog** [comment]
Thanks for pointing me to `command -v`, I totally missed that obvious choice.

I also think that a new warning about "--version" and similar flags would make a lot of sense, as I'm certainly not the only one doing that.

In the specific case of `find`, I'd propose that "--version" simply triggers both warnings, in case there are really some Unix systems where `find --version` doesn't do the expected thing.

**Comment by adrelanos** [comment]
Another false-positive here when `find` is used with the `-files0-from` optoin.

    printf '%s' "${folder_name}" | find -files0-from - -perm /u=s,g=s -print0
